### PR TITLE
Remove Tableau from Salesforce provider

### PR DESCRIPTION
**Pull request by eladkal**
Tableau was split from Salesforce provider a year+ ago in https://github.com/apache/airflow/pull/14030
We kept Tableau in Salesforce to preserve backward compatibility, it's time to remove the deprecated dependency.

This is a breaking change for Salesforce provider.
I didn't place entry in changelog as it caused problems in the past for the automatic release process @potiuk please let me know if you prefer otherwise

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by jhtimmins** [comment]
@potiuk @ashb Since this is a breaking change for the salesforce provider, does that mean we can't merge until 4.0?

**Comment by eladkal** [comment]
We can because this is a provider relase not Airflow core.

File: docs/apache-airflow-providers-salesforce/index.rst
Comment (ashb) [inline docs/apache-airflow-providers-salesforce/index.rst:95]: Is it worth keeping a note here saying "want tableu? It's moved"?

**Review by ashb — APPROVED** [review]
LGTM, I can't remember what changes we need to make now to mark this as needing a major ver bump of the provider when we next release it though.

File: docs/apache-airflow-providers-salesforce/index.rst
Comment (eladkal) [reply docs/apache-airflow-providers-salesforce/index.rst:95]: That is covered by the change log of the provider :) 

1. If someone is currently using Tableau via Salesforce - the user is aware it's deprecated by warnings and will know it's removed by reading the change log.
2. If someone never used Tableau before and searches for tableau provider I want him/her to land in Tableau docs rather than Salesforce docs only to be referred to Tableau provider.

**Review by kaxil — APPROVED** [review]

**Comment by eladkal** [comment]
PR is waiting for discussion on https://lists.apache.org/thread/6ngq79df7op541gfwntspdtsvzlv1cr6 to be finished and probably a voting to pass.

File: generated/provider_dependencies.json
Comment (eladkal) [inline generated/provider_dependencies.json]: @potiuk can you confirm this is right?
wasn't sure with the recent change of removing provider dependencies from core.
Did I miss something else needed?

File: docs/apache-airflow-providers-salesforce/index.rst
Comment (potiuk) [reply docs/apache-airflow-providers-salesforce/index.rst:95]: This is all good in general (but needs a note in CHANGELOG). We have "addional extra" - if somoene wants tableau they should install salesforce provider with `["tableau"]` extra (or install `[tableau]' extra in airflow or install `apache-airflow-providers-extra'. But I agree we should add a note in the chanelog of salesforce. And answering the question of @ashb below - this is described actually in changelog.rst of each provider (kind of preparation to make them fully independent and splittable): 

https://raw.githubusercontent.com/apache/airflow/main/airflow/providers/salesforce/CHANGELOG.rst

```
.. NOTE TO CONTRIBUTORS:
   Please, only add notes to the Changelog just below the "Changelog" header when there are some breaking changes
   and you want to add an explanation to the users on how they are supposed to deal with them.
   The changelog is updated and maintained semi-automatically by release manager.
```

Comment (potiuk) [reply docs/apache-airflow-providers-salesforce/index.rst:95]: Hmm. Looking at the build image failure in the PR. looks strange.

Comment (eladkal) [reply docs/apache-airflow-providers-salesforce/index.rst:95]: added note https://github.com/apache/airflow/pull/23747/commits/19f6aced38e2d76560e283447f1e111cef171be6

File: generated/provider_dependencies.json
Comment (potiuk) [reply generated/provider_dependencies.json]: OK. Got it. This is wrong:

```
    "simple-salesforce>=1.0.0",
    ],
    "cross-providers-deps": []
``` 

Json does not allow comma at the end of the list (after simple-salesforce). This is why "Build CI" image is failing - when you look at the error raised in build CI you will see this:


> #69 1.592       Exception while loading provider dependencies Expecting value: line 555 column 5 (char 11592)

This is the error from json parser complaining about the comma.

This is an interesting one and a good learning/opportunity to improve this part.

1) You should NOT modify this file by hand. It should be updated automatically by pre-commit. I specifically put it into "generated" folder and added a README there to explain this (because json does not support comments, so you cannot really explain it in the file). But I entirely see (and I expected that) that people will attempt to modify it (this is why I put it in the "generated" folder. to indicate you should not modify it.

2) The pre-commits had no chance to work because of json parsing at image build time (inside setup.py). Otherwise you would likely figure out that you should not modify it, because you would see pre-commit failing while regenerating the file and showing you generated diff in the CI. This did not happen - because CI image failed to build because of json parsing error. This is a bit chicken-egg problem if someone modifies the .json in the way that fails parsing.

Proposals how to improve (besides fixing it):

* I can simply parse the .json outside of the setup.py as part of the CI job building the image. Then I could provide a very explicit message "PLEASE DO NOT modify this file".
* I can also run only that `pre-commit` before building the image to regenerate it. This might be nicer because it will not stop the image from building (the regenerated .json file will be ALWAYS good because it is well, generated. Then tthe whole build and test process will continue - but static checks will fail when running the same pre-commit in the "CI" workflow.  - > I think this is best way

Does any of you have any better idea how to signal users they should not modify this file :) @ashb @kaxil @eladkal :). I thought I went a bit above and beyond but even that did not prevent a committer and PMC member to THINK they should modify it  (and the day after it's been merged) :). I am not sure if I have any more ideas :P

Comment (eladkal) [reply generated/provider_dependencies.json]: @potiuk I modified this file only because pre commit asked me to I didnt even know about this file before the pre commit error - this is why I asked you it was strange to me that the error came from pre commit

Comment (eladkal) [reply generated/provider_dependencies.json]: oh wait I read your message wrong.
I think what we have now is OK - I edited it manually because I had issue with pre-commits.
i think we just need to add to the pre-commit message also the sentence "Don't fix manually, run the pre commit to automatically handle this"

Comment (potiuk) [reply generated/provider_dependencies.json]: Yeah. Makes sense

Comment (potiuk) [reply generated/provider_dependencies.json]: https://github.com/apache/airflow/pull/24763

**Comment by potiuk** [comment]
All right - turns that we might have a bit of a problem with building ARM images when cache is not refreshed (or when there are too many changes in Dockerfie/python deps - our machine has 50 minutes time of life and building in parallel 4 ARM images on it "from almost scratch" - I will have to increase the time of life for the ARM instance a bit :)

**Comment by potiuk** [comment]
Fix for that one here: https://github.com/apache/airflow/pull/24765

**Comment by potiuk** [comment]
Rebased to check if the timeout fix for ARM image wil work here :)

**Comment by eladkal** [comment]
Now that lazy consensus passed
https://lists.apache.org/thread/sthhjwd90xnmokyk4m15hcxgwdv38j4j
and https://github.com/apache/airflow/pull/24680 is merged with updated policy
We can proceed to merge this PR.

**Pull request merged**
